velox/row/UnsafeRowDeserializers.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/row/UnsafeRowDeserializers.h"
+
+namespace facebook::velox::row {
+namespace {
+
+inline int64_t getFieldOffset(int64_t nullBitsetWidthInBytes, int32_t index) {
+  return nullBitsetWidthInBytes + UnsafeRow::kFieldWidthBytes * index;
+}
+
+inline bool isNullAt(const uint8_t* memoryAddress, int32_t index) {
+  return bits::isBitSet(memoryAddress, index);
+}
+
+int32_t getTotalStringSize(
+    int32_t columnIdx,
+    int32_t numRows,
+    int64_t fieldOffset,
+    const std::vector<size_t>& offsets,
+    const uint8_t* memoryAddress) {
+  size_t size = 0;
+  for (auto pos = 0; pos < numRows; pos++) {
+    if (isNullAt(memoryAddress + offsets[pos], columnIdx)) {
+      continue;
+    }
+
+    int64_t offsetAndSize =
+        *(int64_t*)(memoryAddress + offsets[pos] + fieldOffset);
+    int32_t length = static_cast<int32_t>(offsetAndSize);
+    if (!StringView::isInline(length)) {
+      size += length;
+    }
+  }
+  return size;
+}
+
+template <TypeKind Kind>
+VectorPtr createFlatVectorFast(
+    const TypePtr& type,
+    int32_t columnIdx,
+    int32_t numRows,
+    int64_t fieldOffset,
+    const std::vector<size_t>& offsets,
+    const uint8_t* memoryAddress,
+    memory::MemoryPool* pool) {
+  using T = typename TypeTraits<Kind>::NativeType;
+  constexpr uint32_t typeWidth = sizeof(T);
+  auto column = BaseVector::create<FlatVector<T>>(type, numRows, pool);
+  auto rawValues = column->template mutableRawValues<uint8_t>();
+  auto shift = __builtin_ctz(typeWidth);
+  for (auto pos = 0; pos < numRows; pos++) {
+    if (!isNullAt(memoryAddress + offsets[pos], columnIdx)) {
+      const uint8_t* srcptr = (memoryAddress + offsets[pos] + fieldOffset);
+      uint8_t* destptr = rawValues + (pos << shift);
```

Comment:
nit: use camel case for variable names. srcptr-> srcPtr, wordoffset -> wordOffset, destptr -> destPtr, ...